[ReflectiveObsidian] iP

ReflectiveObsidian opened the pull request:
### Pyrite

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Pyrite frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4.  let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ReflectiveObsidian committed:
Replaced old greeting

ReflectiveObsidian committed:
Implemented command echoing, with exit word.

ReflectiveObsidian committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

ReflectiveObsidian committed:
Moved to-do list into separate class

ReflectiveObsidian committed:
Added task class

ReflectiveObsidian committed:
Implement task status generation

ReflectiveObsidian committed:
Switch from list of string to list of Task in Pyrite

ReflectiveObsidian committed:
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

ReflectiveObsidian committed:
Add "Here are the tasks in your list:" to task list printing

ReflectiveObsidian committed:
Fixed off-by-1 bug in mark/unmark

ReflectiveObsidian committed:
convert getFullStatusString into toString

ReflectiveObsidian committed:
override toString in Task instead of implementing getFullStatusString()

ReflectiveObsidian committed:
Create 3 types of tasks, ToDo, Deadline, Event

ReflectiveObsidian committed:
Integrate ToDos, Deadlines, Events

ReflectiveObsidian committed:
fix indentation in mark/unmark

ReflectiveObsidian committed:
Fix bug parsing event when order of parameters are swapped

ReflectiveObsidian committed:
Implemented Automated Testing

ReflectiveObsidian committed:
Remove extra horizontal line from bye command

ReflectiveObsidian committed:
Implement exception error handling.

ReflectiveObsidian committed:
Use ArrayList to store tasks

ReflectiveObsidian committed:
Implement delete

ReflectiveObsidian committed:
Update test cases

ReflectiveObsidian committed:
Use enums to track whether task is done or not

ReflectiveObsidian committed:
Make some variables in Pyrite static

ReflectiveObsidian committed:
Add save to file feature

ReflectiveObsidian committed:
Merge branch 'branch-Level-7' into master

ReflectiveObsidian committed:
Store dates as a java.time.LocalDateTime object in Task Objects

ReflectiveObsidian committed:
Merge branch 'branch-Level-8' into master

ReflectiveObsidian committed:
Extract Task List into TaskList class

ReflectiveObsidian committed:
Refactor command handling out of the main loop, using classes Command and Parser

ReflectiveObsidian committed:
Extract the UI class

ReflectiveObsidian committed:
Move attribution for parser logic

ReflectiveObsidian committed:
Remove unused variable

ReflectiveObsidian committed:
Fix bug in input checking during deletion

ReflectiveObsidian committed:
Merge tag 'A-MoreOOP' into master

ReflectiveObsidian committed:
Organize into Packages

ReflectiveObsidian committed:
Merge branch 'add-gradle-support' into master

ReflectiveObsidian committed:
Add dummy tests

ReflectiveObsidian committed:
Add todoTest

ReflectiveObsidian committed:
Add TaskListTest

ReflectiveObsidian committed:
Extend TaskList tests

ReflectiveObsidian committed:
Add parser test

ReflectiveObsidian committed:
Update StatusCommand access parameters

ReflectiveObsidian committed:
Merge branch 'A-JUnit' into master

ReflectiveObsidian committed:
Update build.gradle for jar creation

ReflectiveObsidian committed:
Add JavaDoc comments

ReflectiveObsidian committed:
Comply with Naming coding standard

ReflectiveObsidian committed:
Add find command

ReflectiveObsidian committed:
Merge branch 'branch-A-JavaDoc' into master

ReflectiveObsidian committed:
Merge branch 'branch-A-CodingStandard' into master

# Conflicts:
#	src/main/java/pyrite/TaskList.java

ReflectiveObsidian committed:
Merge branch 'branch-Level-9' into master

# Conflicts:
#	src/main/java/pyrite/TaskList.java

ReflectiveObsidian committed:
Add GUI

ReflectiveObsidian committed:
Merge branch 'branch-Level-10' into master

ReflectiveObsidian committed:
Add and comply to checkstyle

ReflectiveObsidian committed:
Merge branch 'branch-A-CheckStyle' into master

ReflectiveObsidian committed:
Add ability to parse multiple commands per user input

ReflectiveObsidian committed:
Merge branch 'branch-A-Varargs' into master

RunjiaChen commented on src/main/java/pyrite/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+     */
+    public boolean isValidId(int id) {
+        if (id < 0 || id >= this.tasks.size()) {
+            return false;
```
I like how you followed the convention and made boolean variables more clear

RunjiaChen commented on src/main/java/pyrite/Parser.java:
```diff
@@ -0,0 +1,166 @@
+        return id;
+    }
+
+    private static int findCommand(String[] toSearch, String toFind) {
```
Should this be a boolean value instead?

RunjiaChen commented on src/main/java/pyrite/task/Event.java:
```diff
@@ -0,0 +1,38 @@
+                + super.toString()
+                + " (from: "
+                + start.format(DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm"))
+                + " to: "
```
I like how you kept within the character limit per line by inserting line breaks here

RunjiaChen commented on src/main/java/pyrite/Parser.java:
```diff
@@ -0,0 +1,166 @@
+                        return new UnknownCommand(commandString,
+                                "Incomplete pyrite.command.Command. Add start and end dates using '/from' and '/to'.");
+                    }
+                    String fromString;
```
I like how you followed the correct formatting for switch statements.

RunjiaChen commented on src/main/java/pyrite/task/Task.java:
```diff
@@ -0,0 +1,45 @@
+     * Enum to represent the status of a task.
+     */
+    public enum Status {
+        DONE,
```
Should this be made protected or private instead?

RunjiaChen reviewed (commented):
Overall, your code LGTM, with just a few nits to pick.

ReflectiveObsidian committed:
Update build information

The class and method used to launch the program has
changed.

This change means that building the jar file does not produce a valid
program.

Updating the Gradle configuration with the new names creates a working
jar.

ReflectiveObsidian committed:
Update name of built file

ReflectiveObsidian committed:
Use assert feature in StateFile and Task

There are no assertions in the code.

Assertions document assumptions and surface incorrect ones.

Add assertions to points in the code where assumptions are being made.

Several methods in StateFile and Task make assumptions about their input
parameters, which should be documented.

ReflectiveObsidian committed:
Use streams to search for index of string in string array

A for loop is used to search for a string in an array.

This method is less readable because the happy path and intent to find
the index of the first occurring string is not clear.

Use streams to search for the first occurrence of the string instead.

This way, the intent is clear and the default case of returning -1 is
obvious.

ReflectiveObsidian committed:
Reduce nesting

ReflectiveObsidian committed:
Move magic strings into named constants

ReflectiveObsidian committed:
Merge pull request #1 from ReflectiveObsidian/branch-A-Assertions

Use assert feature in StateFile and Task

ReflectiveObsidian committed:
Merge pull request #2 from ReflectiveObsidian/branch-A-CodeQuality

Branch a code quality

ReflectiveObsidian committed:
Merge branch 'master' into branch-A-CodeQuality

ReflectiveObsidian committed:
Merge pull request #3 from ReflectiveObsidian/branch-A-Streams

Use streams to search for index of string in string array

ReflectiveObsidian committed:
Merge branch 'master' into branch-A-Streams

ReflectiveObsidian committed:
Create gradle.yml

ReflectiveObsidian committed:
Merge branch 'master' of https://github.com/ReflectiveObsidian/ip into master

ReflectiveObsidian committed:
Fix regression in find command that resulted in no results

ReflectiveObsidian committed:
Make find command case-insensitive

ReflectiveObsidian committed:
Merge branch 'branch-A-CodeQuality' into master

ReflectiveObsidian committed:
Allow dialog box to expand with text

ReflectiveObsidian committed:
Reduce size of profile picture

ReflectiveObsidian committed:
Change profile picture to circle shape

ReflectiveObsidian committed:
Make window resizable

ReflectiveObsidian committed:
Remove extra newlines from responses

ReflectiveObsidian committed:
Alter phrases

ReflectiveObsidian committed:
Check if start date of event is after the end date

ReflectiveObsidian committed:
Add more testcases

ReflectiveObsidian committed:
Add comments to denote use of Github Copilot in creating test cases

ReflectiveObsidian committed:
Remove references to Duke

ReflectiveObsidian committed:
Fix incorrect empty description test case

ReflectiveObsidian committed:
Fix dialog box text not aligning to right side for user dialog

ReflectiveObsidian committed:
Add product screenshot

ReflectiveObsidian committed:
Update README.md

ReflectiveObsidian committed:
Update README.md

ReflectiveObsidian committed:
Update README.md to include link to user guide

ReflectiveObsidian committed:
Update README.md

ReflectiveObsidian committed:
Update README.md to fix user guide link

ReflectiveObsidian committed:
Update README.md

ReflectiveObsidian committed:
Remove dummy tests

ReflectiveObsidian committed:
Add Javadoc to test cases

ReflectiveObsidian committed:
UnknownCommandTest: Fix import style

ReflectiveObsidian committed:
Add error dialog box

ReflectiveObsidian committed:
Refactor Parser

ReflectiveObsidian committed:
Update Parser test cases

ReflectiveObsidian committed:
Update string access modifier in Parser

ReflectiveObsidian committed:
Remove unused UnknownCommand

ReflectiveObsidian committed:
Add javadocs to test cases

ReflectiveObsidian committed:
Update README.md

ReflectiveObsidian committed:
Add more javadocs

ReflectiveObsidian committed:
Merge branch 'master' of https://github.com/ReflectiveObsidian/ip into master

ReflectiveObsidian committed:
Rename enum

ReflectiveObsidian committed:
Fix wrong import orders

ReflectiveObsidian committed:
Fix incorrect error message

ReflectiveObsidian committed:
Update command list in error message
